### PR TITLE
test: Acceptance tests

### LINKED ISSUE #119: Add end to end tests
### Package

All packages

### Description

Add end to end test using one of the example apps. We've been using [testcafe](https://github.com/DevExpress/testcafe) on other projects with some success, it should be fairly straight forward to set that up.

It would be great if we could, as part of a PR, build and deploy an example app and then run the tests against it. 

If we used the next.js example(would require bringing in to this repo), then we could use now.sh and it's github pull request feature. The other option is using Heroku's review apps with any of our examples.

### PR DESCRIPTION
**Pull request by Joezo**
### What does this PR do?

Adds a simple acceptance test that ensures the create-react-app example renders a list of posts.

The majority of this PR is setting up the CI. The process is as follows:
- PR is opened
- CircleCI runs normal "test" build
- Netlify deploys preview of create-react-app example.
- Netlify sends a request on `deploy-succeeded` to CircleCI telling it to run the job `acceptance-tests` on correct branch, along with a `ACCEPTANCE_URL` parameter that points to the Netlify preview.
- CircleCI runs `npm run test:acceptance:ci`, which uses testcafe and browserstack.

This process makes use Netlify functions, which handily you can trigger from certain Netlify events. We make use of the `deploy-succeeded` event by naming our function file `deploy-succeeded.js`, easy. 

### Related issues

closes #119 

### Checklist

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [x] I have added or updated any relevant documentation
- [x] I have added or updated any relevant tests

### CODE REVIEW

File: .netlify/functions/deploy-succeeded.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    headers: {
+      'Content-Type': 'application/json'
+    },
+    auth: process.env.CIRCLECI_PROJECT_TOKEN
+  }
+
+  const postData = JSON.stringify({
+    build_parameters: {
+      CIRCLE_JOB: 'acceptance-tests',
+      ACCEPTANCE_URL: 'boo'
```
Comment (jackdclark) [inline .netlify/functions/deploy-succeeded.js]: 👻

**Comment by jackdclark** [comment]
This is looking awesome 👏

File: .eslintrc.js [hunk 2]
```diff
@@ -14,7 +14,9 @@ module.exports = {
   ],
   globals: {
     Atomics: 'readonly',
-    SharedArrayBuffer: 'readonly'
+    SharedArrayBuffer: 'readonly',
+    fixture: true,
+    test: true
```
Comment (jackdclark) [inline .eslintrc.js]: We could add [eslint-plugin-testcafe](https://www.npmjs.com/package/eslint-plugin-testcafe) for this

File: README.md [hunk 3]
```diff
@@ -655,3 +655,7 @@ Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/d
 <!-- ALL-CONTRIBUTORS-LIST:END -->
 
 This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
+
+[![BrowserStack](https://p14.zdusercontent.com/attachment/1015988/mg687dwxHqXtriITEf8kxZV3W?token=eyJhbGciOiJkaXIiLCJlbmMiOiJBMTI4Q0JDLUhTMjU2In0..tPLabhhdTeWxyc3TTt-RCg.bmk4nO95zIaYIcNaaDEVtxph9ap6d9X__07O0wPpvgsx5RBYvue1gMxCGhnYcgtQA51YjC5BFCxev9bBGZ0f6wHGr83j_nBID68oZCdgurHQhuZjsBZTotXtVdGDJoGg8KHMvl2qK9_FFlxohxGkPatEyccPXfLxZGGrGhvGnZVs6sFcy5bSevRHwe84yH3y0-PhbwE9HPAqzYsJyjBsSnez3gllgrIqX_7UucPPyAxtESSOaevl3zs6n5EfJ6teaJ3_KhWTmux9Nlk5csiWwvcRcCXp7p14Xln9tBYR64k.-1SqygSW1Ke0iJ-t3ED3SQ)](http://browserstack.com/)
```
Comment (jackdclark) [inline README.md]: It's so big!

File: .eslintrc.js (hunk 2 above)
Comment (Joezo) [reply .eslintrc.js]: Yeah all it does is exactly these two lines so didn't bother.

Comment (jackdclark) [reply .eslintrc.js]: Just means if they update their API then it will stay in line

**Pull request merged**